### PR TITLE
add tests, allow css

### PR DESCRIPTION
**Pull request by ThomasAndrewMacLean**
## What
* convert zero height json to css

## How
*  🧙‍♂️ Magic

## Demo

image
image

### CODE REVIEW

File: packages/zero-height/src/convertToCss.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+describe('@bothrs/zero-height ~ converToCss', () => {
+  test('should return css', () => {
+    const input = {
+      Themes: {
+        Theme1: '#fff',
+        Theme2: '#eee',
+      },
+    }
+
+    const result = converToCss(input)
```
Comment (thgh) [inline packages/zero-height/src/convertToCss.test.ts:12]: Typo
```suggestion
    const result = convertToCss(input)
```

File: packages/zero-height/src/convertToCss.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// 🐉 here be dragons 🐲
+const flattenObject = (obj: any, prefix = '') =>
```
Comment (thgh) [inline packages/zero-height/src/convertToCss.ts:2]: "delimiter" or "glue" could be a better name for the prefix param

File: packages/zero-height/src/convertToCss.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  Object.keys(obj).reduce((acc, k) => {
+    const pre = prefix.length ? prefix + '-' : ''
+    if (typeof obj[k] === 'object')
+      Object.assign(acc, flattenObject(obj[k], pre + k))
+    //@ts-ignore
+    else acc[pre + k] = obj[k]
+    return acc
+  }, {})
```
Comment (thgh) [inline packages/zero-height/src/convertToCss.ts:10]: Another way to resolve the ts ignore:
```suggestion
  }, {} as any)
```

File: packages/zero-height/src/convertToCss.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+export const converToCss = (fixedJSON: Record<string, any>): string => {
+  // the object we receive can be pretty nested OR it can be a font
+  const duplicate = { ...fixedJSON }
```
Comment (thgh) [inline packages/zero-height/src/convertToCss.ts:14]: Why is this needed? (Not seeing mutations)

File: packages/zero-height/src/convertToCss.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  const keys = Object.keys(duplicate)
+  let cssVariables = ':root{'
+  let fonts = ''
+  for (let index = 0; index < keys.length; index++) {
```
Comment (thgh) [inline packages/zero-height/src/convertToCss.ts:19]: Equivalent:
```suggestion
  for (let key in fixedJSON) {
```

**Review by thgh — APPROVED** [review]

File: packages/zero-height/src/convertToCss.ts (hunk 3 above)
Comment (bramvanhoutte) [reply packages/zero-height/src/convertToCss.ts:10]: You could also use a more specific type like `Record<string, any>`, to give us a little useful typing.

I'd also pass the type as a generic argument in the reducer function instead of casting the initial value (But that's more personal preference I think)

File: packages/zero-height/src/convertToCss.test.ts (hunk 1 above)
Comment (thgh) [reply packages/zero-height/src/convertToCss.test.ts]: er zijn nog extra occurrences 😬

**Review by FabianMeul — APPROVED** [review]
